spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/actions/DeleteOrphanFilesSparkAction.java

```diff
@@ -246,12 +247,24 @@ private DeleteOrphanFiles.Result doExecute() {
     List<String> orphanFiles =
         findOrphanFiles(spark(), actualFileIdentDS, validFileIdentDS, prefixMismatchMode);
 
-    Tasks.foreach(orphanFiles)
-        .noRetry()
-        .executeWith(deleteExecutorService)
-        .suppressFailureWhenFinished()
-        .onFailure((file, exc) -> LOG.warn("Failed to delete file: {}", file, exc))
-        .run(deleteFunc::accept);
+    if (deleteFunc == null && table.io() instanceof SupportsBulkOperations) {
+      ((SupportsBulkOperations) table.io()).deleteFiles(orphanFiles);
+
+    } else {
+      if (deleteFunc == null) {
+        LOG.info("Bulk Deletes are not Supported by {}, using non-bulk deletes", table.io());
+        deleteFunc = defaultDelete;
```

Comment:
I don't think it is a good idea to init this variable here. I summarized my thoughts in other actions below. One way to avoid this:

```
if (deleteFunc == null) {
  LOG.info("Table IO does not support bulk operations, using non-bulk deletes");
} else {
  LOG.info("Custom delete function provided, using non-bulk deletes");
}

Tasks.foreach(orphanFiles)
    .noRetry()
    ...
    .run(deleteFunc != null ? deleteFunc::accept : table.io()::deleteFile);
```